### PR TITLE
elf,bpf_test: Improve error checks

### PR DESCRIPTION
**Pull request by dongsupark**
Remove an unnecessary error check in `elfReadVersion()`. The error is already being checked above in the same function, https://github.com/iovisor/gobpf/blob/master/elf/elf.go#L284

We should check for a specific error code `os.ErrInvalid`, instead of comparing error strings with each other.

### CODE REVIEW

File: bpf_test.go [hunk 1]
```diff
@@ -255,7 +255,7 @@ func TestModuleLoadELF(t *testing.T) {
 		t.Fatal("prog is nil")
 	}
 	if err := b.Load(nil); err != nil {
-		if !strings.Contains(err.Error(), "invalid argument") {
```
Comment (schu) [inline bpf_test.go]: I wonder why we ignore the error at all? @iaguis do you remember? Changed with 8f18c857026f0495eed5b00c28f9697548d88519

Comment (iaguis) [reply bpf_test.go]: I don't really know :/

I think we can just remove it.

Comment (dongsupark) [reply bpf_test.go]: @iaguis Removed the check for invalid argument.

**Comment by schu** [comment]
Thanks!

**Pull request merged**